### PR TITLE
Support for the STRUMPACK sparse direct solver and preconditioner [strumpack-dev]

### PR DESCRIPTION
**Pull request by pghysels**
#### Pull request for comments about the integration of `strumpack-dev` into `master`.

- For STRUMPACK, see: https://github.com/pghysels/STRUMPACK.
- Look in `example_build.sh` and `INSTALL.txt` for installation instructions.
- This branch requires `metis`, `scotch`, `parmetis`, `ptscotch`, `blas`, `lapack`, `scalapack`.

#### Test with the new Example 18p:
```
mpirun -np 4 ex18p -m ../data/square-disc.mesh --strumpack
```

#### TODO
- [x] make sure Travis CI and Appveyor tests pass
- [x] add building description in the `INSTALL` file
- [x] add a note in the `CHANGELOG` file
- [x] update website documentation in https://github.com/mfem/web repo:
    - [x] add a label in the [Examples documentation](http://mfem.org/examples/), or maybe group together with SuperLU, see https://github.com/mfem/web/blob/master/src/examples.md
    - [x] mention on the [Features page](http://mfem.org/features/), see https://github.com/mfem/web/blob/master/src/features.md

### CODE REVIEW

File: examples/ex18p.cpp [hunk 1]
```diff
@@ -0,0 +1,328 @@
+//                       MFEM Example 18 - Parallel Version
```
Comment (tzanio) [inline examples/ex18p.cpp]: It will be best if we add the `--strumpack` option to `ex11p` instead of having a separate example. Can we do that?

Comment (pghysels) [inline examples/ex18p.cpp]: Yes, I can do that. Will do it soon.

**Comment by tzanio** [comment]
@pghysels, can you check the edits in the INSTALL and CHANGELOG files as well as [this PR](https://github.com/mfem/web/pull/4) for updating the documentation?  (Feel free to change anything as you see fit.)

I'd like to try to merge this today if possible. @mlstowell, @v-dobrev -- Any objections?

**Comment by pghysels** [comment]
STRUMPACK also depends on ParMetis and Metis.
In defaults.mk STRUMPACK_LIB includes SCOTCH_LIB and SCALAPACK_LIB, but not METIS_LIB. In defaults.mk, SUPERLU_LIB also does not depend include the ParMetis library. But in defaults.cmake, ParMETIS is listed as a requirement for SuperLUDist.

In INSTALL, it says that "STRUMPACK requires the PT-Scotch and Scalapack libraries." But STRUMPACK also requires ParMetis and Metis, just like SuperLUDist.
I just want to make sure you are aware of the dependency of STRUMPACK on (Par)METIS.

**Comment by tzanio** [comment]
@pghysels: ParMetis issues fixed in db368dd1962b449cc0c502f370d96f1e826d8f5c?

**Comment by tzanio** [comment]
Merged in the `next` branch for testing. Thanks @pghysels and @xiaoyeli!

**Pull request merged**